src/contract/deploy.rs

```diff
@@ -72,15 +75,15 @@ where
 {
     type Output = Result<I, DeployError>;
 
-    fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
-        let unpinned = self.get_mut();
-        Pin::new(&mut unpinned.network_id)
+    fn poll(mut self: Pin<&mut Self>, cx: &mut Context) -> Poll<Self::Output> {
+        self.as_mut()
```

Comment:
Interesting that we are now requiring a mutable instance of self.

nit: `as_mut` could be avoided if web3, context get queried outside of the map closure.

Reply:
Yeah, that is since `fn as_mut(&mut Pin<P>) -> Pin<&mut <P as Deref>::Target>`. Its so the borrow checker makes sure that from a single `Pin<P>` you can't get multiple `Pin<&mut <P as Deref>::Target>` which would mean that you have multiple mutable references to the same thing :fearful:

Reply:
The problem if we take it out of the closure, is we `.take()` if from the `Option`, so if the future is not resolved the first time its polled, then it will panic the second time.

Reply:
Ah, you are right. Let's keep it as is then.